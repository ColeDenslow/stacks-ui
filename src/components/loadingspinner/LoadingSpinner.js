import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './LoadingSpinner.css';

function LoadingSpinner(props) {
  const { className, ...otherProps } = props;

  const classNames = cn([
    'stx-loading-spinner',
    className,
  ]);

  return (
    <span
      className={classNames}
      {...otherProps}
    />
  );
}

LoadingSpinner.propTypes = {
  /** Optional className for the component */
  className: PropTypes.string,
};

export default LoadingSpinner;
