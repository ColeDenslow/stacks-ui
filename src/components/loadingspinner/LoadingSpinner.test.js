import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner', () => {
  test('LoadingSpinner renders with basic props', () => {
    const output = mount(<LoadingSpinner
      className='the-class'
      data-additional-prop='additional-prop'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });
});
