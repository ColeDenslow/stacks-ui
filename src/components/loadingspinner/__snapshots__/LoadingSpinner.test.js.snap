// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner LoadingSpinner renders with basic props 1`] = `
<LoadingSpinner
  className="the-class"
  data-additional-prop="additional-prop"
>
  <span
    className="stx-loading-spinner the-class"
    data-additional-prop="additional-prop"
  />
</LoadingSpinner>
`;
